File: Data Structures & Algorithms/three-integer-sum/submission-2.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int n=nums.size();
        vector<vector<int>> ans;
        for(int i=0;i<n;i++){
            if(i>0){
                if(nums[i-1]==nums[i]) continue;
            }
            int j=i+1;
            int k=n-1;
            while(j<k){
                int target=nums[i]+nums[j]+nums[k];
                if(target==0) {
                    ans.push_back({nums[i],nums[j],nums[k]});
                    j++;
                    k--;
                }
                if(target>0){
                    k--;
                    while(nums[k]==nums[k+1] && k>j) k--;
                } 
                if(target<0){
                    j++;
                    while(nums[j-1]==nums[j] && j<n) j++;
                }
            }
        }
        return ans;   
    }
};

File: Data Structures & Algorithms/car-fleet/submission-2.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        pair = []
        for i in range(len(position)):
            pair.append((position[i], speed[i]))
        pair.sort(reverse=True)
        stack = []
        for p, s in pair:
            time = (target - p) / s
            if stack and time <= stack[-1]:
                continue
            stack.append(time)
        return len(stack)